app/components/viral/modal_component.html.erb

```diff
@@ -0,0 +1,73 @@
+<turbo-frame id="modal">
+  <div data-controller="viral--modal-component">
+    <!-- Main modal -->
+    <div
+      data-viral--modal-component-target="modal"
+      tabindex="-1"
+      aria-hidden="true"
+      class="
+        fixed
+        top-0
+        left-0
+        right-0
+        z-50
+        hidden
+        w-full
+        p-4
+        overflow-x-hidden
+        overflow-y-auto
+        md:inset-0
+        h-[calc(100%-1rem)]
+        max-h-full
+      "
+    >
+      <div class="relative w-full max-w-2xl max-h-full">
+        <!-- Modal content -->
+        <div class="relative bg-white rounded-lg shadow dark:bg-gray-700">
+          <!-- Modal header -->
+          <div
+            id="Viral-Modal-Header"
+            class="
+              flex
+              items-start
+              justify-between
+              p-4
+              border-b
+              rounded-t
+              dark:border-gray-600
+            "
+          >
+            <h3 class="text-xl font-semibold text-gray-900 dark:text-white">
+              <%= title %>
+            </h3>
+            <button
+              type="button"
+              class="
+                text-gray-400
+                bg-transparent
+                hover:bg-gray-200
+                hover:text-gray-900
+                rounded-lg
+                text-sm
+                p-1.5
+                ml-auto
+                inline-flex
+                items-center
+                dark:hover:bg-gray-600
+                dark:hover:text-white
+              "
+              data-action="click->viral--modal-component#close"
+            >
+              <%= viral_icon(name: "x_mark", classes: "w-5 h-5") %>
+              <span class="sr-only">Close modal</span>
+            </button>
+          </div>
+          <!-- Modal body -->
+          <div id="Viral-Modal-Body" class="p-6 space-y-6">
```

Comment:
**issue (blocking):** Text is not optimized for dark mode.  To help you can see the dark mode in LookBook preview, there is a toggle in the menu bar.

Reply:
Please see adf30595bd5fa82d8e5b6830338b0017ead26a97.